### PR TITLE
Move JHttpFactory::getHttp() into try catch and add error handling

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Move any usage of  `JHttpFactory::getHttp()` into try catch and add error handling

### Testing Instructions

Review.

Confirm that you can still uses a feed.
Confirm that recaptcha still works.

### Expected result

everything still works

### Actual result

everything works but without error handling

### Documentation Changes Required

None.

### CODE REVIEW

File: libraries/joomla/feed/factory.php [hunk 1]
```diff
@@ -44,19 +44,25 @@ public function getFeed($uri)
...
 
-			$connector 	= JHttpFactory::getHttp($options);
-			$feed 		= $connector->get($uri);
+			try
+			{
+				$response = JHttpFactory::getHttp($options)->get($uri);
+			}
+			catch (RuntimeException $exception)
+			{
+				throw new RuntimeException('Unable to open the feed.');
```
Comment (mbabker) [inline libraries/joomla/feed/factory.php]: If you're just going to catch and throw a new Exception with no extra handling, there is no point in catching this.  Plus, it makes error handling more difficult if you don't pass the caught Exception into the newly thrown Exception (basically `throw new RuntimeException('Unable to open the feed.', $exception->getCode(), $exception);`)

Comment (zero-24) [reply libraries/joomla/feed/factory.php]: What about throw new RuntimeException($e->getMessage()); ?

File: libraries/joomla/oauth1/client.php [hunk 2]
```diff
@@ -70,10 +70,21 @@ public function __construct(Registry $options = null, JHttp $client = null, JInp
...
+		try
+		{
+			$http = JHttpFactory::getHttp($options);
+		}
+		catch (RuntimeException $exception)
+		{
+			$http = null;
+		}
+
+		$this->client = isset($client) ? $client : $http;
```
Comment (mbabker) [inline libraries/joomla/oauth1/client.php]: You would need to add checks if the client isn't a `JHttp` instance.  Here it'd be better to just not catch it at all, especially as child classes basically expect this to always be a `JHttp` instance.

File: libraries/joomla/feed/factory.php (hunk 1 above)
Comment (mbabker) [reply libraries/joomla/feed/factory.php]: The caught Exception here is going to have more error information, including the stack trace from where that Exception is thrown from.  So by not including that Exception in the re-throw, you're actively stripping debugging information.

File: libraries/joomla/oauth1/client.php (hunk 2 above)
Comment (zero-24) [reply libraries/joomla/oauth1/client.php]: I'm not sure what you mean here? Do you want me to add a check like `is the passed client a instance of JHttp` before doing the try catch?

Comment (mbabker) [reply libraries/joomla/oauth1/client.php]: No.  Basically revert the changes in this file.

File: libraries/joomla/feed/factory.php (hunk 1 above)
Comment (zero-24) [reply libraries/joomla/feed/factory.php]: done with the last two commits.

File: libraries/joomla/oauth1/client.php (hunk 2 above)
Comment (zero-24) [reply libraries/joomla/oauth1/client.php]: done.

**Review by mbabker — APPROVED** [review]

**Comment by zero-24** [comment]
Can we get tests that captcha is still loading and using of a feed still works?

**Comment by zero-24** [comment]
drone is happy now. So we can start testing here?

**Comment by zero-24** [comment]
I'm going to trash that PR and redo it from scratch.. something did broken this PR.

**Pull request closed without merging**